src/queries/SearchResultQuery.ts

```diff
@@ -17,21 +24,32 @@ interface SearchResultQueryInterface extends SearchResultQueryParams {
 
 class SearchResultQuery extends BaseQuery<SearchResult> implements SearchResultQueryInterface {
   keyword?: string;
+  url?: string;
+  filterAdUrlWord?: string;
   userId: number;
   filterQuery: { [index: string]: any };
+  filterUrlCharacter?: string;
+  filterUrlCharacterAtLeast: number;
 
-  constructor(searchResultQueryParams: SearchResultQueryParams) {
-    super({ ...searchResultQueryParams, databaseModal: dbClient.searchResult });
+  constructor(userId: number, queryParams: SearchResultQueryParams) {
+    super({ ...queryParams, databaseModal: dbClient.searchResult });
 
-    this.keyword = searchResultQueryParams.keyword;
-    this.userId = searchResultQueryParams.userId;
+    this.userId = userId;
+    this.keyword = queryParams.filters?.keyword;
+    this.url = queryParams.filters?.url;
+    this.filterAdUrlWord = queryParams.filters?.filterAdUrlWord;
+    this.filterUrlCharacter = queryParams.filters?.filterUrlCharacter;
+    this.filterUrlCharacterAtLeast = queryParams.filters?.filterUrlCharacterAtLeast || 1;
```

Comment:
These grow fast 🙈 We don't need to specify them like that, right?

```suggestion
    this.filters = queryParams.filters;
```